sqs/data_mappings.bal

```diff
@@ -290,7 +290,6 @@ isolated function read(string path) returns @tainted json|FileReadFailed {
         io:ReadableCharacterChannel readableChannel = new(readableByteChannel, "UTF8");
         var result = readableChannel.readJson();
         if (result is error) {
-            FileReadFailed? err = closeReadableChannel(readableChannel);
```

Comment:
Is it fine not to close the channel?

Reply:
Even if we do, having this in `data_mappings` seems wrong. We need to return the error and handle it in the client or listener class.

Reply:
We will close the channel

Reply:
Noticed it is not called within the code. So that's fine. We need to clean up the code in a separate PR. 

```
isolated function read(string path) returns @tainted json|FileReadFailed {
    io:ReadableByteChannel|error readableByteChannel = io:openReadableFile(path);
    if (readableByteChannel is io:ReadableByteChannel) {
        io:ReadableCharacterChannel readableChannel = new(readableByteChannel, "UTF8");
        var result = readableChannel.readJson();
        if (result is error) {
            FileReadFailed? err = closeReadableChannel(readableChannel);
            return error FileReadFailed(FILE_READ_FAILED_MSG, result);
        } else {
            FileReadFailed? err = closeReadableChannel(readableChannel);
            if (err is error) {
                return error FileReadFailed(FILE_READ_FAILED_MSG, err);
            } else {
                return result;
            }
        }
    } else {
        return error FileReadFailed(FILE_READ_FAILED_MSG, readableByteChannel);
    }
}

isolated function closeReadableChannel(io:ReadableCharacterChannel readableChannel) returns FileReadFailed? {
    var result = readableChannel.close();
    if (result is error) {
        return error FileReadFailed(CLOSE_CHARACTER_STREAM_FAILED_MSG, result);
    }
}
```